### DISCUSSION TITLE
Como em um DynamicForm, carregar itens em um PoCombo e posicionar em um ítem desejado?

### DISCUSSION THREAD

**Discussion by andredourado** [po-ui/po-angular#1375] [Angular]
Estou utilizando o componente DynamicForm. E um dos campo é um PoCombo. Estou implementando uma dependêncioa entre Estado e Cidade. 

Na inclusão ok, pois o conteúdo de cidades é carregado para o estado selescionado, porém na edição, o campo de cidade já vem filtrado para a cidade que consta no banco de dados. E quando posiciono no combo de cidades, as demais cidades do estado não estão carregadas.

Como poderia carregar todas as cidades e apenas selecionar a cidade cadastrada. Que seria o procedimento convencional de um select no html.

Agradeço qualquer auxílio.

**Comment by wsteixeira** [comment]
Olá @andredourado,

Segue um exemplo de uma forma que eu uso [https://stackblitz.com/edit/po-ui-6594k9?file=src%2Fapp%2Fapp.component.ts](https://stackblitz.com/edit/po-ui-6594k9?file=src%2Fapp%2Fapp.component.ts). Espero que ajude!

![image](https://user-images.githubusercontent.com/36740857/186292669-afab16c7-b69d-4a76-9c4f-69bf5fed2be6.png)

**Reply by andredourado** [reply]
Wanderley muito obrigado pela resposta. A questão não é carregar ou filtrar. A questão é que na edição do registro. O componente da cidade utiliza o endpoint para filtrar apenas o que foi previamente informado. Então eu tenho que entrar novamente no estado selecionar o estado novamente para carregar a lista de cidades. Não estou conseguindo que posicione na cidade cadastrada e que tenha a lista de cidades disponível. Você teria algum exemplo que fosse uma edição de registro cadastrado?

**Reply by wsteixeira** [reply]
Olá @andredourado,

Entenda esta parte do exemplo como o retorno da API que carregou o formulário, que você chamou de "previamente informado":
```
    this.value = {
      state: 1,
      city: 208
    }
```
Isso vai disparar o `ngDoCheck()`, que vai ajustar a nova lista de cidades toda vez que o campo `state` for alterado, inclusive na carga inicial:
```
  ngDoCheck() {
    const valueChanges = this.valueDiffers.diff(this.value);

    if (valueChanges) {
      const { state } = this.value;

      // Definir filtro de busca dos documentos conforme fornecedor informado
      if (state) {
        this.appSearchService.setFilterParams(
          { stateId: state, _expand: 'state' }
        );
      }
    }
  }
```
Seria possível você montar um exemplo no [StackBlitz](https://stackblitz.com/edit/po-ui), para facilitar o entendimento e nossa ajuda?

OBS: A primeira vez pode demorar um pouco, porque a API fica em `stand by` no Heroku.
